frontend/plugins/highlight-search.js

```diff
@@ -29,49 +29,37 @@ export default (context, inject) => {
     return `<span class="highlight-text">${htmlText(text)}</span>`;
   }
 
-  const regexFromTerm = function (term) {
-    let q = term.replace(/[-[\]{}()*+?.,\\/^$|#\s]/g, "");
-    return new RegExp(q, "gi");
-  };
-
   const escapeRegExp = function (text) {
     return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
   };
 
-  const highlightSearch = function (query, text) {
-    const escapedText = htmlText(text);
-    if (!query) {
-      return text;
-    }
+  function sortByLength(keywords) {
+    return (keywords || []).sort((a, b) => b.length - a.length);
+  }
 
-    return escapedText.replace(
-      regexFromTerm(query),
-      (match) => `<span class="highlight-text">${match}</span>`
-    );
-  };
+  function createPattern(value) {
+    return `([^a-zA-ZÀ-ÿ\u00f1\u00d1]|^)${escapeRegExp(value)}`;
+  }
+  function createRegExp(pattern) {
+    return new RegExp(pattern, "gmi");
+  }
 
-  const highlightKeywords = function (text, keywords) {
-    const sortedKeywords = ([...keywords] || []).sort(
-      (a, b) => b.length - a.length
+  function replaceText(regex, text) {
+    return htmlText(text).replace(regex, (matched) =>
+      htmlHighlightText(matched)
     );
```

Comment:
Code cleanning : the deepest functions should appears after their implementation 

=> sortByLength/createPattern/replaceText should appears after their implementations in highlightKeywords function